gems/aws-sdk-core/lib/aws-sdk-core/xml/doc_builder.rb

```diff
@@ -6,7 +6,7 @@ class DocBuilder
       # @option options [String] :pad ('')
       # @option options [String] :indent ('')
       def initialize(options = {})
-        @target = options[:target] || ''
+        @target = options[:target] || String.new
```

Comment:
Why String.new here?  I assume because it needs to be mutable - if so, add a comment.